src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +781,262 @@ public function in_default_dir( $src ) {
+					__( 'Invalid strategy `%1$s` defined for `%2$s` during script registration.' ),
+					$strategy,
+					$handle
+				),
+				'6.3.0'
+			);
+
+			return '';
+		}
+
+		return $strategy;
+	}
+
+	/**
+	 * Check if the strategy passed is a non-blocking strategy.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $strategy The strategy to check.
+	 * @return bool True if $strategy is one of the delayed strategy.
+	 */
+	private function is_non_blocking_strategy( $strategy ) {
+		$delayed_strategies = array( 'defer', 'async' );
+
+		return in_array( $strategy, $delayed_strategies, true );
+	}
+
+	/**
+	 * Check if a script has a non standalone inline script associated with it.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle   The script handle.
+	 * @param string $position Position of the inline script.
+	 *
+	 * @return bool True if script present. False if empty.
+	 */
+	private function has_non_standalone_inline_script( $handle, $position ) {
+		$non_standalone_script_key = $position . '-non-standalone';
+		$non_standalone_script     = $this->get_data( $handle, $non_standalone_script_key );
+
+		return ! empty( $non_standalone_script );
+	}
+
+	/**
+	 * Check if all of a scripts dependents are deferrable, which is required to maintain execution order.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle  The script handle.
+	 * @param array  $checked Optional. An array of already checked script handles, used to avoid recursive loops.
+	 * @return bool True if all dependents are deferrable, false otherwise.
+	 */
+	private function has_only_deferrable_dependents( $handle, $checked = array() ) {
+		// If this node was already checked, this script can be deferred and the branch ends.
+		if ( in_array( $handle, $checked, true ) ) {
+			return true;
+		}
+
+		$checked[]  = $handle;
+		$dependents = $this->get_dependents( $handle );
+
+		// If there are no dependents remaining to consider, the script can be deferred.
+		if ( empty( $dependents ) ) {
+			return true;
+		}
+
+		// Consider each dependent and check if it is deferrable.
+		foreach ( $dependents as $dependent ) {
+			// If the dependent script is not using the defer or async strategy, no script in the chain is deferrable.
+			$strategy = $this->get_intended_strategy( $dependent );
+			if ( ! $this->is_non_blocking_strategy( $strategy ) ) {
+				return false;
+			}
+
+			// If the dependent script has a non-standalone inline script in the 'before' position associated with it, do not defer.
+			if ( $this->has_non_standalone_inline_script( $dependent, 'before' ) ) {
+				return false;
+			}
+
+			// Recursively check all dependents.
+			if ( ! $this->has_only_deferrable_dependents( $dependent, $checked ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the best eligible loading strategy for a script.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string  $handle The registered handle of the script.
+	 * @return string $strategy return the final strategy.
```

Comment:
```suggestion
	 * @return string|null $strategy return the final strategy.
```
As BC it return null. WDYT?

Reply:
I don't see where this previously returned `null`. Can you clarify this @mukeshpanchal27? Thanks!

Reply:
@mukeshpanchal27 this method never existed prior, and only returns a string (strategy or empty). I do not think this needs to be effected to be `string|null`